### DISCUSSION TITLE
Idea: Woke shield

### DISCUSSION THREAD

**Discussion by cognitivegears** [get-woke/woke#134] [Ideas]
Was thinking about whether `woke` could have a shield (https://shields.io or similar) - it would be cool to have a `[woke| 0 findings]` button available for repos.  I found this as to implementation: https://github.com/badges/shields/blob/master/doc/TUTORIAL.md

I haven't looked too much into it yet, but maybe possible for github repos using Actions to display their number of findings, etc?

**Comment by caitlinelfring** [comment]
This seems like something that would require some server-side component to use shields like this. Otherwise a basic badge for a repos woke GitHub action job would be easy to do, any repo can do that themselves. 

Not sure I see much value in server side functionality right now though

**Reply by cognitivegears** [reply]
Totally agree about server side functionality. However, my thought was to do this with `woke-action` by adding an option to publish the result to github-pages as a json file. Similar to the actions [Deploy to gethub pages](https://github.com/marketplace/actions/deploy-to-github-pages ), [Actions GH Pages](https://github.com/peaceiris/actions-gh-pages) etc. Then, from there shields.io could use it as an "API" - see [Utilize GitHub Pages as JSON API](https://testfairy.com/blog/utilize-github-pages-as-json-api/) Going this route, I think the changes would include:
* I'd add back in that pull request for the option to create a single json object from `woke`
* Modify `woke-action` to optionally publish that json file to github pages
* Create the PR / feature on shields.io to use that "API" for a new `woke` badge

Thoughts?

**Reply by cognitivegears** [reply]
Btw - if you are happy with the approach I can do the changes/PR's for this one if you want, sounds like fun :)

**Reply by caitlinelfring** [reply]
I don't think I would want to modify the `woke-action` to support this, I'm not sure I see much value in it as a quick github search doesn't show many open source repos using woke. 

If this is something that interests you and you want to play around with it, feel free to create a demo of it under your own account and we can see about moving it under the `get-woke` org if it makes sense to do so.

I also am not sure you really need to generate a single json object for this. If you only really care about the number of findings, the action could just be as simple as `echo "{\"findings_count\": $(woke -o json | wc -l)}" > woke-findings.json` since the shield only cares about the number. It could link to the woke action which would display the details of the findings. 

There's also already a github action badge that shows passing/failing which might be good enough. 

[![woke](https://github.com/get-woke/woke/actions/workflows/woke.yml/badge.svg)](https://github.com/get-woke/woke/actions/workflows/woke.yml)

Generally, this seems a little overly-complicated for the value-add, but feel free to have a go at it within your own account!
